--- a/leds-lp5569.c
+++ b/leds-lp5569.c
@@ -1319,6 +1321,11 @@ static int lp5569_remove(struct i2c_clie
 	return 0;
 }
 
+static void lp5569_remove_void(struct i2c_client *client)
+{
+	lp5569_remove(client);
+}
+
 static const struct i2c_device_id lp5569_id[] = {
 	{ "lp5569",  LP5569 },
 	{ }
@@ -1341,7 +1348,7 @@ static struct i2c_driver lp5569_driver =
 		.of_match_table = of_match_ptr(of_lp5569_leds_match),
 	},
 	.probe		= lp5569_probe,
-	.remove		= lp5569_remove,
+	.remove		= lp5569_remove_void,
 	.id_table	= lp5569_id,
 };
 
