--- a/drivers/net/wireless/ath/ath10k/mac.c
+++ b/drivers/net/wireless/ath/ath10k/mac.c
@@ -24,7 +24,6 @@
 #include "wmi-tlv.h"
 #include "wmi-ops.h"
 #include "wow.h"
-#include "leds.h"
 
 /*********/
 /* Rates */
@@ -5530,7 +5529,7 @@ static int ath10k_mac_set_txbf_conf(stru
 					 ar->wmi.vdev_param->txbf, value);
 }
 
-static void ath10k_update_vif_offload(struct ieee80211_hw *hw,
+static int ath10k_update_vif_offload(struct ieee80211_hw *hw,
 				      struct ieee80211_vif *vif)
 {
 	struct ath10k_vif *arvif = (void *)vif->drv_priv;
@@ -5552,6 +5551,8 @@ static void ath10k_update_vif_offload(st
 		ath10k_warn(ar, "failed to set vdev %i TX encapsulation: %d\n",
 			    arvif->vdev_id, ret);
 	}
+
+	return ret;
 }
 
 /*
